// tw-daterange/dist/DateRangePicker.es.js
import ft, { useState as ae, useEffect as Ke } from "react";
var b = {}, It = {
  get exports() {
    return b;
  },
  set exports(r) {
    b = r;
  }
}, ne = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ze;
function qt() {
  if (Ze)
    return ne;
  Ze = 1;
  var r = ft, e = Symbol.for("react.element"), t = Symbol.for("react.fragment"), n = Object.prototype.hasOwnProperty, o = r.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, i = { key: !0, ref: !0, __self: !0, __source: !0 };
  function s(v, c, g) {
    var f, h = {}, D = null, _ = null;
    g !== void 0 && (D = "" + g), c.key !== void 0 && (D = "" + c.key), c.ref !== void 0 && (_ = c.ref);
    for (f in c)
      n.call(c, f) && !i.hasOwnProperty(f) && (h[f] = c[f]);
    if (v && v.defaultProps)
      for (f in c = v.defaultProps, c)
        h[f] === void 0 && (h[f] = c[f]);
    return { $$typeof: e, type: v, key: D, ref: _, props: h, _owner: o.current };
  }
  return ne.Fragment = t, ne.jsx = s, ne.jsxs = s, ne;
}
var oe = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var et;
function Bt() {
  return et || (et = 1, process.env.NODE_ENV !== "production" && function() {
    var r = ft, e = Symbol.for("react.element"), t = Symbol.for("react.portal"), n = Symbol.for("react.fragment"), o = Symbol.for("react.strict_mode"), i = Symbol.for("react.profiler"), s = Symbol.for("react.provider"), v = Symbol.for("react.context"), c = Symbol.for("react.forward_ref"), g = Symbol.for("react.suspense"), f = Symbol.for("react.suspense_list"), h = Symbol.for("react.memo"), D = Symbol.for("react.lazy"), _ = Symbol.for("react.offscreen"), P = Symbol.iterator, d = "@@iterator";
    function C(a) {
      if (a === null || typeof a != "object")
        return null;
      var u = P && a[P] || a[d];
      return typeof u == "function" ? u : null;
    }
    var O = r.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function E(a) {
      {
        for (var u = arguments.length, l = new Array(u > 1 ? u - 1 : 0), m = 1; m < u; m++)
          l[m - 1] = arguments[m];
        A("error", a, l);
      }
    }
    function A(a, u, l) {
      {
        var m = O.ReactDebugCurrentFrame, p = m.getStackAddendum();
        p !== "" && (u += "%s", l = l.concat([p]));
        var S = l.map(function(y) {
          return String(y);
        });
        S.unshift("Warning: " + u), Function.prototype.apply.call(console[a], console, S);
      }
    }
    var I = !1, ue = !1, F = !1, V = !1, te = !1, Q;
    Q = Symbol.for("react.module.reference");
    function J(a) {
      return !!(typeof a == "string" || typeof a == "function" || a === n || a === i || te || a === o || a === g || a === f || V || a === _ || I || ue || F || typeof a == "object" && a !== null && (a.$$typeof === D || a.$$typeof === h || a.$$typeof === s || a.$$typeof === v || a.$$typeof === c || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      a.$$typeof === Q || a.getModuleId !== void 0));
    }
    function Te(a, u, l) {
      var m = a.displayName;
      if (m)
        return m;
      var p = u.displayName || u.name || "";
      return p !== "" ? l + "(" + p + ")" : l;
    }
    function le(a) {
      return a.displayName || "Context";
    }
    function $(a) {
      if (a == null)
        return null;
      if (typeof a.tag == "number" && E("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof a == "function")
        return a.displayName || a.name || null;
      if (typeof a == "string")
        return a;
      switch (a) {
        case n:
          return "Fragment";
        case t:
          return "Portal";
        case i:
          return "Profiler";
        case o:
          return "StrictMode";
        case g:
          return "Suspense";
        case f:
          return "SuspenseList";
      }
      if (typeof a == "object")
        switch (a.$$typeof) {
          case v:
            var u = a;
            return le(u) + ".Consumer";
          case s:
            var l = a;
            return le(l._context) + ".Provider";
          case c:
            return Te(a, a.render, "ForwardRef");
          case h:
            var m = a.displayName || null;
            return m !== null ? m : $(a.type) || "Memo";
          case D: {
            var p = a, S = p._payload, y = p._init;
            try {
              return $(y(S));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var q = Object.assign, Y = 0, L, G, Re, We, Ye, Ne, je;
    function Fe() {
    }
    Fe.__reactDisabledLog = !0;
    function wt() {
      {
        if (Y === 0) {
          L = console.log, G = console.info, Re = console.warn, We = console.error, Ye = console.group, Ne = console.groupCollapsed, je = console.groupEnd;
          var a = {
            configurable: !0,
            enumerable: !0,
            value: Fe,
            writable: !0
          };
          Object.defineProperties(console, {
            info: a,
            log: a,
            warn: a,
            error: a,
            group: a,
            groupCollapsed: a,
            groupEnd: a
          });
        }
        Y++;
      }
    }
    function yt() {
      {
        if (Y--, Y === 0) {
          var a = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: q({}, a, {
              value: L
            }),
            info: q({}, a, {
              value: G
            }),
            warn: q({}, a, {
              value: Re
            }),
            error: q({}, a, {
              value: We
            }),
            group: q({}, a, {
              value: Ye
            }),
            groupCollapsed: q({}, a, {
              value: Ne
            }),
            groupEnd: q({}, a, {
              value: je
            })
          });
        }
        Y < 0 && E("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var xe = O.ReactCurrentDispatcher, Ce;
    function ce(a, u, l) {
      {
        if (Ce === void 0)
          try {
            throw Error();
          } catch (p) {
            var m = p.stack.trim().match(/\n( *(at )?)/);
            Ce = m && m[1] || "";
          }
        return `
` + Ce + a;
      }
    }
    var _e = !1, fe;
    {
      var bt = typeof WeakMap == "function" ? WeakMap : Map;
      fe = new bt();
    }
    function Ue(a, u) {
      if (!a || _e)
        return "";
      {
        var l = fe.get(a);
        if (l !== void 0)
          return l;
      }
      var m;
      _e = !0;
      var p = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var S;
      S = xe.current, xe.current = null, wt();
      try {
        if (u) {
          var y = function() {
            throw Error();
          };
          if (Object.defineProperty(y.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(y, []);
            } catch (B) {
              m = B;
            }
            Reflect.construct(a, [], y);
          } else {
            try {
              y.call();
            } catch (B) {
              m = B;
            }
            a.call(y.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (B) {
            m = B;
          }
          a();
        }
      } catch (B) {
        if (B && m && typeof B.stack == "string") {
          for (var w = B.stack.split(`
`), N = m.stack.split(`
`), R = w.length - 1, W = N.length - 1; R >= 1 && W >= 0 && w[R] !== N[W]; )
            W--;
          for (; R >= 1 && W >= 0; R--, W--)
            if (w[R] !== N[W]) {
              if (R !== 1 || W !== 1)
                do
                  if (R--, W--, W < 0 || w[R] !== N[W]) {
                    var U = `
` + w[R].replace(" at new ", " at ");
                    return a.displayName && U.includes("<anonymous>") && (U = U.replace("<anonymous>", a.displayName)), typeof a == "function" && fe.set(a, U), U;
                  }
                while (R >= 1 && W >= 0);
              break;
            }
        }
      } finally {
        _e = !1, xe.current = S, yt(), Error.prepareStackTrace = p;
      }
      var K = a ? a.displayName || a.name : "", ze = K ? ce(K) : "";
      return typeof a == "function" && fe.set(a, ze), ze;
    }
    function pt(a, u, l) {
      return Ue(a, !1);
    }
    function Dt(a) {
      var u = a.prototype;
      return !!(u && u.isReactComponent);
    }
    function de(a, u, l) {
      if (a == null)
        return "";
      if (typeof a == "function")
        return Ue(a, Dt(a));
      if (typeof a == "string")
        return ce(a);
      switch (a) {
        case g:
          return ce("Suspense");
        case f:
          return ce("SuspenseList");
      }
      if (typeof a == "object")
        switch (a.$$typeof) {
          case c:
            return pt(a.render);
          case h:
            return de(a.type, u, l);
          case D: {
            var m = a, p = m._payload, S = m._init;
            try {
              return de(S(p), u, l);
            } catch {
            }
          }
        }
      return "";
    }
    var me = Object.prototype.hasOwnProperty, $e = {}, Le = O.ReactDebugCurrentFrame;
    function ve(a) {
      if (a) {
        var u = a._owner, l = de(a.type, a._source, u ? u.type : null);
        Le.setExtraStackFrame(l);
      } else
        Le.setExtraStackFrame(null);
    }
    function Tt(a, u, l, m, p) {
      {
        var S = Function.call.bind(me);
        for (var y in a)
          if (S(a, y)) {
            var w = void 0;
            try {
              if (typeof a[y] != "function") {
                var N = Error((m || "React class") + ": " + l + " type `" + y + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof a[y] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw N.name = "Invariant Violation", N;
              }
              w = a[y](u, y, m, l, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (R) {
              w = R;
            }
            w && !(w instanceof Error) && (ve(p), E("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", m || "React class", l, y, typeof w), ve(null)), w instanceof Error && !(w.message in $e) && ($e[w.message] = !0, ve(p), E("Failed %s type: %s", l, w.message), ve(null));
          }
      }
    }
    var xt = Array.isArray;
    function Oe(a) {
      return xt(a);
    }
    function Ct(a) {
      {
        var u = typeof Symbol == "function" && Symbol.toStringTag, l = u && a[Symbol.toStringTag] || a.constructor.name || "Object";
        return l;
      }
    }
    function _t(a) {
      try {
        return Ae(a), !1;
      } catch {
        return !0;
      }
    }
    function Ae(a) {
      return "" + a;
    }
    function Ie(a) {
      if (_t(a))
        return E("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", Ct(a)), Ae(a);
    }
    var re = O.ReactCurrentOwner, Ot = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, qe, Be, Ee;
    Ee = {};
    function Et(a) {
      if (me.call(a, "ref")) {
        var u = Object.getOwnPropertyDescriptor(a, "ref").get;
        if (u && u.isReactWarning)
          return !1;
      }
      return a.ref !== void 0;
    }
    function St(a) {
      if (me.call(a, "key")) {
        var u = Object.getOwnPropertyDescriptor(a, "key").get;
        if (u && u.isReactWarning)
          return !1;
      }
      return a.key !== void 0;
    }
    function kt(a, u) {
      if (typeof a.ref == "string" && re.current && u && re.current.stateNode !== u) {
        var l = $(re.current.type);
        Ee[l] || (E('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', $(re.current.type), a.ref), Ee[l] = !0);
      }
    }
    function Mt(a, u) {
      {
        var l = function() {
          qe || (qe = !0, E("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", u));
        };
        l.isReactWarning = !0, Object.defineProperty(a, "key", {
          get: l,
          configurable: !0
        });
      }
    }
    function Pt(a, u) {
      {
        var l = function() {
          Be || (Be = !0, E("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", u));
        };
        l.isReactWarning = !0, Object.defineProperty(a, "ref", {
          get: l,
          configurable: !0
        });
      }
    }
    var Rt = function(a, u, l, m, p, S, y) {
      var w = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: e,
        // Built-in properties that belong on the element
        type: a,
        key: u,
        ref: l,
        props: y,
        // Record the component responsible for creating this element.
        _owner: S
      };
      return w._store = {}, Object.defineProperty(w._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(w, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: m
      }), Object.defineProperty(w, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: p
      }), Object.freeze && (Object.freeze(w.props), Object.freeze(w)), w;
    };
    function Wt(a, u, l, m, p) {
      {
        var S, y = {}, w = null, N = null;
        l !== void 0 && (Ie(l), w = "" + l), St(u) && (Ie(u.key), w = "" + u.key), Et(u) && (N = u.ref, kt(u, p));
        for (S in u)
          me.call(u, S) && !Ot.hasOwnProperty(S) && (y[S] = u[S]);
        if (a && a.defaultProps) {
          var R = a.defaultProps;
          for (S in R)
            y[S] === void 0 && (y[S] = R[S]);
        }
        if (w || N) {
          var W = typeof a == "function" ? a.displayName || a.name || "Unknown" : a;
          w && Mt(y, W), N && Pt(y, W);
        }
        return Rt(a, w, N, p, m, re.current, y);
      }
    }
    var Se = O.ReactCurrentOwner, He = O.ReactDebugCurrentFrame;
    function z(a) {
      if (a) {
        var u = a._owner, l = de(a.type, a._source, u ? u.type : null);
        He.setExtraStackFrame(l);
      } else
        He.setExtraStackFrame(null);
    }
    var ke;
    ke = !1;
    function Me(a) {
      return typeof a == "object" && a !== null && a.$$typeof === e;
    }
    function Ve() {
      {
        if (Se.current) {
          var a = $(Se.current.type);
          if (a)
            return `

Check the render method of \`` + a + "`.";
        }
        return "";
      }
    }
    function Yt(a) {
      {
        if (a !== void 0) {
          var u = a.fileName.replace(/^.*[\\\/]/, ""), l = a.lineNumber;
          return `

Check your code at ` + u + ":" + l + ".";
        }
        return "";
      }
    }
    var Ge = {};
    function Nt(a) {
      {
        var u = Ve();
        if (!u) {
          var l = typeof a == "string" ? a : a.displayName || a.name;
          l && (u = `

Check the top-level render call using <` + l + ">.");
        }
        return u;
      }
    }
    function Xe(a, u) {
      {
        if (!a._store || a._store.validated || a.key != null)
          return;
        a._store.validated = !0;
        var l = Nt(u);
        if (Ge[l])
          return;
        Ge[l] = !0;
        var m = "";
        a && a._owner && a._owner !== Se.current && (m = " It was passed a child from " + $(a._owner.type) + "."), z(a), E('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', l, m), z(null);
      }
    }
    function Qe(a, u) {
      {
        if (typeof a != "object")
          return;
        if (Oe(a))
          for (var l = 0; l < a.length; l++) {
            var m = a[l];
            Me(m) && Xe(m, u);
          }
        else if (Me(a))
          a._store && (a._store.validated = !0);
        else if (a) {
          var p = C(a);
          if (typeof p == "function" && p !== a.entries)
            for (var S = p.call(a), y; !(y = S.next()).done; )
              Me(y.value) && Xe(y.value, u);
        }
      }
    }
    function jt(a) {
      {
        var u = a.type;
        if (u == null || typeof u == "string")
          return;
        var l;
        if (typeof u == "function")
          l = u.propTypes;
        else if (typeof u == "object" && (u.$$typeof === c || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        u.$$typeof === h))
          l = u.propTypes;
        else
          return;
        if (l) {
          var m = $(u);
          Tt(l, a.props, "prop", m, a);
        } else if (u.PropTypes !== void 0 && !ke) {
          ke = !0;
          var p = $(u);
          E("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", p || "Unknown");
        }
        typeof u.getDefaultProps == "function" && !u.getDefaultProps.isReactClassApproved && E("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function Ft(a) {
      {
        for (var u = Object.keys(a.props), l = 0; l < u.length; l++) {
          var m = u[l];
          if (m !== "children" && m !== "key") {
            z(a), E("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", m), z(null);
            break;
          }
        }
        a.ref !== null && (z(a), E("Invalid attribute `ref` supplied to `React.Fragment`."), z(null));
      }
    }
    function Je(a, u, l, m, p, S) {
      {
        var y = J(a);
        if (!y) {
          var w = "";
          (a === void 0 || typeof a == "object" && a !== null && Object.keys(a).length === 0) && (w += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var N = Yt(p);
          N ? w += N : w += Ve();
          var R;
          a === null ? R = "null" : Oe(a) ? R = "array" : a !== void 0 && a.$$typeof === e ? (R = "<" + ($(a.type) || "Unknown") + " />", w = " Did you accidentally export a JSX literal instead of a component?") : R = typeof a, E("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", R, w);
        }
        var W = Wt(a, u, l, p, S);
        if (W == null)
          return W;
        if (y) {
          var U = u.children;
          if (U !== void 0)
            if (m)
              if (Oe(U)) {
                for (var K = 0; K < U.length; K++)
                  Qe(U[K], a);
                Object.freeze && Object.freeze(U);
              } else
                E("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Qe(U, a);
        }
        return a === n ? Ft(W) : jt(W), W;
      }
    }
    function Ut(a, u, l) {
      return Je(a, u, l, !0);
    }
    function $t(a, u, l) {
      return Je(a, u, l, !1);
    }
    var Lt = $t, At = Ut;
    oe.Fragment = n, oe.jsx = Lt, oe.jsxs = At;
  }()), oe;
}
(function(r) {
  process.env.NODE_ENV === "production" ? r.exports = qt() : r.exports = Bt();
})(It);
function k(r) {
  if (r === null || r === !0 || r === !1)
    return NaN;
  var e = Number(r);
  return isNaN(e) ? e : e < 0 ? Math.ceil(e) : Math.floor(e);
}
function x(r, e) {
  if (e.length < r)
    throw new TypeError(r + " argument" + (r > 1 ? "s" : "") + " required, but only " + e.length + " present");
}
function he(r) {
  return typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? he = function(t) {
    return typeof t;
  } : he = function(t) {
    return t && typeof Symbol == "function" && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t;
  }, he(r);
}
function M(r) {
  x(1, arguments);
  var e = Object.prototype.toString.call(r);
  return r instanceof Date || he(r) === "object" && e === "[object Date]" ? new Date(r.getTime()) : typeof r == "number" || e === "[object Number]" ? new Date(r) : ((typeof r == "string" || e === "[object String]") && typeof console < "u" && (console.warn("Starting with v2.0.0-beta.1 date-fns doesn't accept strings as date arguments. Please use `parseISO` to parse strings. See: https://github.com/date-fns/date-fns/blob/master/docs/upgradeGuide.md#string-arguments"), console.warn(new Error().stack)), new Date(NaN));
}
function dt(r, e) {
  x(2, arguments);
  var t = M(r), n = k(e);
  return isNaN(n) ? new Date(NaN) : (n && t.setDate(t.getDate() + n), t);
}
function mt(r, e) {
  x(2, arguments);
  var t = M(r), n = k(e);
  if (isNaN(n))
    return new Date(NaN);
  if (!n)
    return t;
  var o = t.getDate(), i = new Date(t.getTime());
  i.setMonth(t.getMonth() + n + 1, 0);
  var s = i.getDate();
  return o >= s ? i : (t.setFullYear(i.getFullYear(), i.getMonth(), o), t);
}
function ge(r) {
  return typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? ge = function(t) {
    return typeof t;
  } : ge = function(t) {
    return t && typeof Symbol == "function" && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t;
  }, ge(r);
}
function j(r, e) {
  if (x(2, arguments), !e || ge(e) !== "object")
    return new Date(NaN);
  var t = e.years ? k(e.years) : 0, n = e.months ? k(e.months) : 0, o = e.weeks ? k(e.weeks) : 0, i = e.days ? k(e.days) : 0, s = e.hours ? k(e.hours) : 0, v = e.minutes ? k(e.minutes) : 0, c = e.seconds ? k(e.seconds) : 0, g = M(r), f = n || t ? mt(g, n + t * 12) : g, h = i || o ? dt(f, i + o * 7) : f, D = v + s * 60, _ = c + D * 60, P = _ * 1e3, d = new Date(h.getTime() + P);
  return d;
}
function Ht(r, e) {
  x(2, arguments);
  var t = M(r).getTime(), n = k(e);
  return new Date(t + n);
}
var Vt = {};
function De() {
  return Vt;
}
function Gt(r) {
  var e = new Date(Date.UTC(r.getFullYear(), r.getMonth(), r.getDate(), r.getHours(), r.getMinutes(), r.getSeconds(), r.getMilliseconds()));
  return e.setUTCFullYear(r.getFullYear()), r.getTime() - e.getTime();
}
function tt(r) {
  x(1, arguments);
  var e = M(r);
  return e.setHours(0, 0, 0, 0), e;
}
function Z(r, e) {
  x(2, arguments);
  var t = tt(r), n = tt(e);
  return t.getTime() === n.getTime();
}
function we(r) {
  return typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? we = function(t) {
    return typeof t;
  } : we = function(t) {
    return t && typeof Symbol == "function" && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t;
  }, we(r);
}
function Xt(r) {
  return x(1, arguments), r instanceof Date || we(r) === "object" && Object.prototype.toString.call(r) === "[object Date]";
}
function Qt(r) {
  if (x(1, arguments), !Xt(r) && typeof r != "number")
    return !1;
  var e = M(r);
  return !isNaN(Number(e));
}
function Jt(r) {
  x(1, arguments);
  var e = M(r), t = e.getMonth();
  return e.setFullYear(e.getFullYear(), t + 1, 0), e.setHours(23, 59, 59, 999), e;
}
function zt(r) {
  x(1, arguments);
  var e = M(r);
  return e.setDate(1), e.setHours(0, 0, 0, 0), e;
}
function Kt(r, e) {
  x(2, arguments);
  var t = k(e);
  return Ht(r, -t);
}
var Zt = 864e5;
function er(r) {
  x(1, arguments);
  var e = M(r), t = e.getTime();
  e.setUTCMonth(0, 1), e.setUTCHours(0, 0, 0, 0);
  var n = e.getTime(), o = t - n;
  return Math.floor(o / Zt) + 1;
}
function be(r) {
  x(1, arguments);
  var e = 1, t = M(r), n = t.getUTCDay(), o = (n < e ? 7 : 0) + n - e;
  return t.setUTCDate(t.getUTCDate() - o), t.setUTCHours(0, 0, 0, 0), t;
}
function vt(r) {
  x(1, arguments);
  var e = M(r), t = e.getUTCFullYear(), n = new Date(0);
  n.setUTCFullYear(t + 1, 0, 4), n.setUTCHours(0, 0, 0, 0);
  var o = be(n), i = new Date(0);
  i.setUTCFullYear(t, 0, 4), i.setUTCHours(0, 0, 0, 0);
  var s = be(i);
  return e.getTime() >= o.getTime() ? t + 1 : e.getTime() >= s.getTime() ? t : t - 1;
}
function tr(r) {
  x(1, arguments);
  var e = vt(r), t = new Date(0);
  t.setUTCFullYear(e, 0, 4), t.setUTCHours(0, 0, 0, 0);
  var n = be(t);
  return n;
}
var rr = 6048e5;
function ar(r) {
  x(1, arguments);
  var e = M(r), t = be(e).getTime() - tr(e).getTime();
  return Math.round(t / rr) + 1;
}
function pe(r, e) {
  var t, n, o, i, s, v, c, g;
  x(1, arguments);
  var f = De(), h = k((t = (n = (o = (i = e == null ? void 0 : e.weekStartsOn) !== null && i !== void 0 ? i : e == null || (s = e.locale) === null || s === void 0 || (v = s.options) === null || v === void 0 ? void 0 : v.weekStartsOn) !== null && o !== void 0 ? o : f.weekStartsOn) !== null && n !== void 0 ? n : (c = f.locale) === null || c === void 0 || (g = c.options) === null || g === void 0 ? void 0 : g.weekStartsOn) !== null && t !== void 0 ? t : 0);
  if (!(h >= 0 && h <= 6))
    throw new RangeError("weekStartsOn must be between 0 and 6 inclusively");
  var D = M(r), _ = D.getUTCDay(), P = (_ < h ? 7 : 0) + _ - h;
  return D.setUTCDate(D.getUTCDate() - P), D.setUTCHours(0, 0, 0, 0), D;
}
function ht(r, e) {
  var t, n, o, i, s, v, c, g;
  x(1, arguments);
  var f = M(r), h = f.getUTCFullYear(), D = De(), _ = k((t = (n = (o = (i = e == null ? void 0 : e.firstWeekContainsDate) !== null && i !== void 0 ? i : e == null || (s = e.locale) === null || s === void 0 || (v = s.options) === null || v === void 0 ? void 0 : v.firstWeekContainsDate) !== null && o !== void 0 ? o : D.firstWeekContainsDate) !== null && n !== void 0 ? n : (c = D.locale) === null || c === void 0 || (g = c.options) === null || g === void 0 ? void 0 : g.firstWeekContainsDate) !== null && t !== void 0 ? t : 1);
  if (!(_ >= 1 && _ <= 7))
    throw new RangeError("firstWeekContainsDate must be between 1 and 7 inclusively");
  var P = new Date(0);
  P.setUTCFullYear(h + 1, 0, _), P.setUTCHours(0, 0, 0, 0);
  var d = pe(P, e), C = new Date(0);
  C.setUTCFullYear(h, 0, _), C.setUTCHours(0, 0, 0, 0);
  var O = pe(C, e);
  return f.getTime() >= d.getTime() ? h + 1 : f.getTime() >= O.getTime() ? h : h - 1;
}
function nr(r, e) {
  var t, n, o, i, s, v, c, g;
  x(1, arguments);
  var f = De(), h = k((t = (n = (o = (i = e == null ? void 0 : e.firstWeekContainsDate) !== null && i !== void 0 ? i : e == null || (s = e.locale) === null || s === void 0 || (v = s.options) === null || v === void 0 ? void 0 : v.firstWeekContainsDate) !== null && o !== void 0 ? o : f.firstWeekContainsDate) !== null && n !== void 0 ? n : (c = f.locale) === null || c === void 0 || (g = c.options) === null || g === void 0 ? void 0 : g.firstWeekContainsDate) !== null && t !== void 0 ? t : 1), D = ht(r, e), _ = new Date(0);
  _.setUTCFullYear(D, 0, h), _.setUTCHours(0, 0, 0, 0);
  var P = pe(_, e);
  return P;
}
var or = 6048e5;
function ir(r, e) {
  x(1, arguments);
  var t = M(r), n = pe(t, e).getTime() - nr(t, e).getTime();
  return Math.round(n / or) + 1;
}
function T(r, e) {
  for (var t = r < 0 ? "-" : "", n = Math.abs(r).toString(); n.length < e; )
    n = "0" + n;
  return t + n;
}
var sr = {
  // Year
  y: function(e, t) {
    var n = e.getUTCFullYear(), o = n > 0 ? n : 1 - n;
    return T(t === "yy" ? o % 100 : o, t.length);
  },
  // Month
  M: function(e, t) {
    var n = e.getUTCMonth();
    return t === "M" ? String(n + 1) : T(n + 1, 2);
  },
  // Day of the month
  d: function(e, t) {
    return T(e.getUTCDate(), t.length);
  },
  // AM or PM
  a: function(e, t) {
    var n = e.getUTCHours() / 12 >= 1 ? "pm" : "am";
    switch (t) {
      case "a":
      case "aa":
        return n.toUpperCase();
      case "aaa":
        return n;
      case "aaaaa":
        return n[0];
      case "aaaa":
      default:
        return n === "am" ? "a.m." : "p.m.";
    }
  },
  // Hour [1-12]
  h: function(e, t) {
    return T(e.getUTCHours() % 12 || 12, t.length);
  },
  // Hour [0-23]
  H: function(e, t) {
    return T(e.getUTCHours(), t.length);
  },
  // Minute
  m: function(e, t) {
    return T(e.getUTCMinutes(), t.length);
  },
  // Second
  s: function(e, t) {
    return T(e.getUTCSeconds(), t.length);
  },
  // Fraction of second
  S: function(e, t) {
    var n = t.length, o = e.getUTCMilliseconds(), i = Math.floor(o * Math.pow(10, n - 3));
    return T(i, t.length);
  }
};
const H = sr;
var ee = {
  am: "am",
  pm: "pm",
  midnight: "midnight",
  noon: "noon",
  morning: "morning",
  afternoon: "afternoon",
  evening: "evening",
  night: "night"
}, ur = {
  // Era
  G: function(e, t, n) {
    var o = e.getUTCFullYear() > 0 ? 1 : 0;
    switch (t) {
      case "G":
      case "GG":
      case "GGG":
        return n.era(o, {
          width: "abbreviated"
        });
      case "GGGGG":
        return n.era(o, {
          width: "narrow"
        });
      case "GGGG":
      default:
        return n.era(o, {
          width: "wide"
        });
    }
  },
  // Year
  y: function(e, t, n) {
    if (t === "yo") {
      var o = e.getUTCFullYear(), i = o > 0 ? o : 1 - o;
      return n.ordinalNumber(i, {
        unit: "year"
      });
    }
    return H.y(e, t);
  },
  // Local week-numbering year
  Y: function(e, t, n, o) {
    var i = ht(e, o), s = i > 0 ? i : 1 - i;
    if (t === "YY") {
      var v = s % 100;
      return T(v, 2);
    }
    return t === "Yo" ? n.ordinalNumber(s, {
      unit: "year"
    }) : T(s, t.length);
  },
  // ISO week-numbering year
  R: function(e, t) {
    var n = vt(e);
    return T(n, t.length);
  },
  // Extended year. This is a single number designating the year of this calendar system.
  // The main difference between `y` and `u` localizers are B.C. years:
  // | Year | `y` | `u` |
  // |------|-----|-----|
  // | AC 1 |   1 |   1 |
  // | BC 1 |   1 |   0 |
  // | BC 2 |   2 |  -1 |
  // Also `yy` always returns the last two digits of a year,
  // while `uu` pads single digit years to 2 characters and returns other years unchanged.
  u: function(e, t) {
    var n = e.getUTCFullYear();
    return T(n, t.length);
  },
  // Quarter
  Q: function(e, t, n) {
    var o = Math.ceil((e.getUTCMonth() + 1) / 3);
    switch (t) {
      case "Q":
        return String(o);
      case "QQ":
        return T(o, 2);
      case "Qo":
        return n.ordinalNumber(o, {
          unit: "quarter"
        });
      case "QQQ":
        return n.quarter(o, {
          width: "abbreviated",
          context: "formatting"
        });
      case "QQQQQ":
        return n.quarter(o, {
          width: "narrow",
          context: "formatting"
        });
      case "QQQQ":
      default:
        return n.quarter(o, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Stand-alone quarter
  q: function(e, t, n) {
    var o = Math.ceil((e.getUTCMonth() + 1) / 3);
    switch (t) {
      case "q":
        return String(o);
      case "qq":
        return T(o, 2);
      case "qo":
        return n.ordinalNumber(o, {
          unit: "quarter"
        });
      case "qqq":
        return n.quarter(o, {
          width: "abbreviated",
          context: "standalone"
        });
      case "qqqqq":
        return n.quarter(o, {
          width: "narrow",
          context: "standalone"
        });
      case "qqqq":
      default:
        return n.quarter(o, {
          width: "wide",
          context: "standalone"
        });
    }
  },
  // Month
  M: function(e, t, n) {
    var o = e.getUTCMonth();
    switch (t) {
      case "M":
      case "MM":
        return H.M(e, t);
      case "Mo":
        return n.ordinalNumber(o + 1, {
          unit: "month"
        });
      case "MMM":
        return n.month(o, {
          width: "abbreviated",
          context: "formatting"
        });
      case "MMMMM":
        return n.month(o, {
          width: "narrow",
          context: "formatting"
        });
      case "MMMM":
      default:
        return n.month(o, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Stand-alone month
  L: function(e, t, n) {
    var o = e.getUTCMonth();
    switch (t) {
      case "L":
        return String(o + 1);
      case "LL":
        return T(o + 1, 2);
      case "Lo":
        return n.ordinalNumber(o + 1, {
          unit: "month"
        });
      case "LLL":
        return n.month(o, {
          width: "abbreviated",
          context: "standalone"
        });
      case "LLLLL":
        return n.month(o, {
          width: "narrow",
          context: "standalone"
        });
      case "LLLL":
      default:
        return n.month(o, {
          width: "wide",
          context: "standalone"
        });
    }
  },
  // Local week of year
  w: function(e, t, n, o) {
    var i = ir(e, o);
    return t === "wo" ? n.ordinalNumber(i, {
      unit: "week"
    }) : T(i, t.length);
  },
  // ISO week of year
  I: function(e, t, n) {
    var o = ar(e);
    return t === "Io" ? n.ordinalNumber(o, {
      unit: "week"
    }) : T(o, t.length);
  },
  // Day of the month
  d: function(e, t, n) {
    return t === "do" ? n.ordinalNumber(e.getUTCDate(), {
      unit: "date"
    }) : H.d(e, t);
  },
  // Day of year
  D: function(e, t, n) {
    var o = er(e);
    return t === "Do" ? n.ordinalNumber(o, {
      unit: "dayOfYear"
    }) : T(o, t.length);
  },
  // Day of week
  E: function(e, t, n) {
    var o = e.getUTCDay();
    switch (t) {
      case "E":
      case "EE":
      case "EEE":
        return n.day(o, {
          width: "abbreviated",
          context: "formatting"
        });
      case "EEEEE":
        return n.day(o, {
          width: "narrow",
          context: "formatting"
        });
      case "EEEEEE":
        return n.day(o, {
          width: "short",
          context: "formatting"
        });
      case "EEEE":
      default:
        return n.day(o, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Local day of week
  e: function(e, t, n, o) {
    var i = e.getUTCDay(), s = (i - o.weekStartsOn + 8) % 7 || 7;
    switch (t) {
      case "e":
        return String(s);
      case "ee":
        return T(s, 2);
      case "eo":
        return n.ordinalNumber(s, {
          unit: "day"
        });
      case "eee":
        return n.day(i, {
          width: "abbreviated",
          context: "formatting"
        });
      case "eeeee":
        return n.day(i, {
          width: "narrow",
          context: "formatting"
        });
      case "eeeeee":
        return n.day(i, {
          width: "short",
          context: "formatting"
        });
      case "eeee":
      default:
        return n.day(i, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Stand-alone local day of week
  c: function(e, t, n, o) {
    var i = e.getUTCDay(), s = (i - o.weekStartsOn + 8) % 7 || 7;
    switch (t) {
      case "c":
        return String(s);
      case "cc":
        return T(s, t.length);
      case "co":
        return n.ordinalNumber(s, {
          unit: "day"
        });
      case "ccc":
        return n.day(i, {
          width: "abbreviated",
          context: "standalone"
        });
      case "ccccc":
        return n.day(i, {
          width: "narrow",
          context: "standalone"
        });
      case "cccccc":
        return n.day(i, {
          width: "short",
          context: "standalone"
        });
      case "cccc":
      default:
        return n.day(i, {
          width: "wide",
          context: "standalone"
        });
    }
  },
  // ISO day of week
  i: function(e, t, n) {
    var o = e.getUTCDay(), i = o === 0 ? 7 : o;
    switch (t) {
      case "i":
        return String(i);
      case "ii":
        return T(i, t.length);
      case "io":
        return n.ordinalNumber(i, {
          unit: "day"
        });
      case "iii":
        return n.day(o, {
          width: "abbreviated",
          context: "formatting"
        });
      case "iiiii":
        return n.day(o, {
          width: "narrow",
          context: "formatting"
        });
      case "iiiiii":
        return n.day(o, {
          width: "short",
          context: "formatting"
        });
      case "iiii":
      default:
        return n.day(o, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // AM or PM
  a: function(e, t, n) {
    var o = e.getUTCHours(), i = o / 12 >= 1 ? "pm" : "am";
    switch (t) {
      case "a":
      case "aa":
        return n.dayPeriod(i, {
          width: "abbreviated",
          context: "formatting"
        });
      case "aaa":
        return n.dayPeriod(i, {
          width: "abbreviated",
          context: "formatting"
        }).toLowerCase();
      case "aaaaa":
        return n.dayPeriod(i, {
          width: "narrow",
          context: "formatting"
        });
      case "aaaa":
      default:
        return n.dayPeriod(i, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // AM, PM, midnight, noon
  b: function(e, t, n) {
    var o = e.getUTCHours(), i;
    switch (o === 12 ? i = ee.noon : o === 0 ? i = ee.midnight : i = o / 12 >= 1 ? "pm" : "am", t) {
      case "b":
      case "bb":
        return n.dayPeriod(i, {
          width: "abbreviated",
          context: "formatting"
        });
      case "bbb":
        return n.dayPeriod(i, {
          width: "abbreviated",
          context: "formatting"
        }).toLowerCase();
      case "bbbbb":
        return n.dayPeriod(i, {
          width: "narrow",
          context: "formatting"
        });
      case "bbbb":
      default:
        return n.dayPeriod(i, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // in the morning, in the afternoon, in the evening, at night
  B: function(e, t, n) {
    var o = e.getUTCHours(), i;
    switch (o >= 17 ? i = ee.evening : o >= 12 ? i = ee.afternoon : o >= 4 ? i = ee.morning : i = ee.night, t) {
      case "B":
      case "BB":
      case "BBB":
        return n.dayPeriod(i, {
          width: "abbreviated",
          context: "formatting"
        });
      case "BBBBB":
        return n.dayPeriod(i, {
          width: "narrow",
          context: "formatting"
        });
      case "BBBB":
      default:
        return n.dayPeriod(i, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Hour [1-12]
  h: function(e, t, n) {
    if (t === "ho") {
      var o = e.getUTCHours() % 12;
      return o === 0 && (o = 12), n.ordinalNumber(o, {
        unit: "hour"
      });
    }
    return H.h(e, t);
  },
  // Hour [0-23]
  H: function(e, t, n) {
    return t === "Ho" ? n.ordinalNumber(e.getUTCHours(), {
      unit: "hour"
    }) : H.H(e, t);
  },
  // Hour [0-11]
  K: function(e, t, n) {
    var o = e.getUTCHours() % 12;
    return t === "Ko" ? n.ordinalNumber(o, {
      unit: "hour"
    }) : T(o, t.length);
  },
  // Hour [1-24]
  k: function(e, t, n) {
    var o = e.getUTCHours();
    return o === 0 && (o = 24), t === "ko" ? n.ordinalNumber(o, {
      unit: "hour"
    }) : T(o, t.length);
  },
  // Minute
  m: function(e, t, n) {
    return t === "mo" ? n.ordinalNumber(e.getUTCMinutes(), {
      unit: "minute"
    }) : H.m(e, t);
  },
  // Second
  s: function(e, t, n) {
    return t === "so" ? n.ordinalNumber(e.getUTCSeconds(), {
      unit: "second"
    }) : H.s(e, t);
  },
  // Fraction of second
  S: function(e, t) {
    return H.S(e, t);
  },
  // Timezone (ISO-8601. If offset is 0, output is always `'Z'`)
  X: function(e, t, n, o) {
    var i = o._originalDate || e, s = i.getTimezoneOffset();
    if (s === 0)
      return "Z";
    switch (t) {
      case "X":
        return at(s);
      case "XXXX":
      case "XX":
        return X(s);
      case "XXXXX":
      case "XXX":
      default:
        return X(s, ":");
    }
  },
  // Timezone (ISO-8601. If offset is 0, output is `'+00:00'` or equivalent)
  x: function(e, t, n, o) {
    var i = o._originalDate || e, s = i.getTimezoneOffset();
    switch (t) {
      case "x":
        return at(s);
      case "xxxx":
      case "xx":
        return X(s);
      case "xxxxx":
      case "xxx":
      default:
        return X(s, ":");
    }
  },
  // Timezone (GMT)
  O: function(e, t, n, o) {
    var i = o._originalDate || e, s = i.getTimezoneOffset();
    switch (t) {
      case "O":
      case "OO":
      case "OOO":
        return "GMT" + rt(s, ":");
      case "OOOO":
      default:
        return "GMT" + X(s, ":");
    }
  },
  // Timezone (specific non-location)
  z: function(e, t, n, o) {
    var i = o._originalDate || e, s = i.getTimezoneOffset();
    switch (t) {
      case "z":
      case "zz":
      case "zzz":
        return "GMT" + rt(s, ":");
      case "zzzz":
      default:
        return "GMT" + X(s, ":");
    }
  },
  // Seconds timestamp
  t: function(e, t, n, o) {
    var i = o._originalDate || e, s = Math.floor(i.getTime() / 1e3);
    return T(s, t.length);
  },
  // Milliseconds timestamp
  T: function(e, t, n, o) {
    var i = o._originalDate || e, s = i.getTime();
    return T(s, t.length);
  }
};
function rt(r, e) {
  var t = r > 0 ? "-" : "+", n = Math.abs(r), o = Math.floor(n / 60), i = n % 60;
  if (i === 0)
    return t + String(o);
  var s = e || "";
  return t + String(o) + s + T(i, 2);
}
function at(r, e) {
  if (r % 60 === 0) {
    var t = r > 0 ? "-" : "+";
    return t + T(Math.abs(r) / 60, 2);
  }
  return X(r, e);
}
function X(r, e) {
  var t = e || "", n = r > 0 ? "-" : "+", o = Math.abs(r), i = T(Math.floor(o / 60), 2), s = T(o % 60, 2);
  return n + i + t + s;
}
const lr = ur;
var nt = function(e, t) {
  switch (e) {
    case "P":
      return t.date({
        width: "short"
      });
    case "PP":
      return t.date({
        width: "medium"
      });
    case "PPP":
      return t.date({
        width: "long"
      });
    case "PPPP":
    default:
      return t.date({
        width: "full"
      });
  }
}, gt = function(e, t) {
  switch (e) {
    case "p":
      return t.time({
        width: "short"
      });
    case "pp":
      return t.time({
        width: "medium"
      });
    case "ppp":
      return t.time({
        width: "long"
      });
    case "pppp":
    default:
      return t.time({
        width: "full"
      });
  }
}, cr = function(e, t) {
  var n = e.match(/(P+)(p+)?/) || [], o = n[1], i = n[2];
  if (!i)
    return nt(e, t);
  var s;
  switch (o) {
    case "P":
      s = t.dateTime({
        width: "short"
      });
      break;
    case "PP":
      s = t.dateTime({
        width: "medium"
      });
      break;
    case "PPP":
      s = t.dateTime({
        width: "long"
      });
      break;
    case "PPPP":
    default:
      s = t.dateTime({
        width: "full"
      });
      break;
  }
  return s.replace("{{date}}", nt(o, t)).replace("{{time}}", gt(i, t));
}, fr = {
  p: gt,
  P: cr
};
const dr = fr;
var mr = ["D", "DD"], vr = ["YY", "YYYY"];
function hr(r) {
  return mr.indexOf(r) !== -1;
}
function gr(r) {
  return vr.indexOf(r) !== -1;
}
function ot(r, e, t) {
  if (r === "YYYY")
    throw new RangeError("Use `yyyy` instead of `YYYY` (in `".concat(e, "`) for formatting years to the input `").concat(t, "`; see: https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md"));
  if (r === "YY")
    throw new RangeError("Use `yy` instead of `YY` (in `".concat(e, "`) for formatting years to the input `").concat(t, "`; see: https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md"));
  if (r === "D")
    throw new RangeError("Use `d` instead of `D` (in `".concat(e, "`) for formatting days of the month to the input `").concat(t, "`; see: https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md"));
  if (r === "DD")
    throw new RangeError("Use `dd` instead of `DD` (in `".concat(e, "`) for formatting days of the month to the input `").concat(t, "`; see: https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md"));
}
var wr = {
  lessThanXSeconds: {
    one: "less than a second",
    other: "less than {{count}} seconds"
  },
  xSeconds: {
    one: "1 second",
    other: "{{count}} seconds"
  },
  halfAMinute: "half a minute",
  lessThanXMinutes: {
    one: "less than a minute",
    other: "less than {{count}} minutes"
  },
  xMinutes: {
    one: "1 minute",
    other: "{{count}} minutes"
  },
  aboutXHours: {
    one: "about 1 hour",
    other: "about {{count}} hours"
  },
  xHours: {
    one: "1 hour",
    other: "{{count}} hours"
  },
  xDays: {
    one: "1 day",
    other: "{{count}} days"
  },
  aboutXWeeks: {
    one: "about 1 week",
    other: "about {{count}} weeks"
  },
  xWeeks: {
    one: "1 week",
    other: "{{count}} weeks"
  },
  aboutXMonths: {
    one: "about 1 month",
    other: "about {{count}} months"
  },
  xMonths: {
    one: "1 month",
    other: "{{count}} months"
  },
  aboutXYears: {
    one: "about 1 year",
    other: "about {{count}} years"
  },
  xYears: {
    one: "1 year",
    other: "{{count}} years"
  },
  overXYears: {
    one: "over 1 year",
    other: "over {{count}} years"
  },
  almostXYears: {
    one: "almost 1 year",
    other: "almost {{count}} years"
  }
}, yr = function(e, t, n) {
  var o, i = wr[e];
  return typeof i == "string" ? o = i : t === 1 ? o = i.one : o = i.other.replace("{{count}}", t.toString()), n != null && n.addSuffix ? n.comparison && n.comparison > 0 ? "in " + o : o + " ago" : o;
};
const br = yr;
function Pe(r) {
  return function() {
    var e = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {}, t = e.width ? String(e.width) : r.defaultWidth, n = r.formats[t] || r.formats[r.defaultWidth];
    return n;
  };
}
var pr = {
  full: "EEEE, MMMM do, y",
  long: "MMMM do, y",
  medium: "MMM d, y",
  short: "MM/dd/yyyy"
}, Dr = {
  full: "h:mm:ss a zzzz",
  long: "h:mm:ss a z",
  medium: "h:mm:ss a",
  short: "h:mm a"
}, Tr = {
  full: "{{date}} 'at' {{time}}",
  long: "{{date}} 'at' {{time}}",
  medium: "{{date}}, {{time}}",
  short: "{{date}}, {{time}}"
}, xr = {
  date: Pe({
    formats: pr,
    defaultWidth: "full"
  }),
  time: Pe({
    formats: Dr,
    defaultWidth: "full"
  }),
  dateTime: Pe({
    formats: Tr,
    defaultWidth: "full"
  })
};
const Cr = xr;
var _r = {
  lastWeek: "'last' eeee 'at' p",
  yesterday: "'yesterday at' p",
  today: "'today at' p",
  tomorrow: "'tomorrow at' p",
  nextWeek: "eeee 'at' p",
  other: "P"
}, Or = function(e, t, n, o) {
  return _r[e];
};
const Er = Or;
function ie(r) {
  return function(e, t) {
    var n = t != null && t.context ? String(t.context) : "standalone", o;
    if (n === "formatting" && r.formattingValues) {
      var i = r.defaultFormattingWidth || r.defaultWidth, s = t != null && t.width ? String(t.width) : i;
      o = r.formattingValues[s] || r.formattingValues[i];
    } else {
      var v = r.defaultWidth, c = t != null && t.width ? String(t.width) : r.defaultWidth;
      o = r.values[c] || r.values[v];
    }
    var g = r.argumentCallback ? r.argumentCallback(e) : e;
    return o[g];
  };
}
var Sr = {
  narrow: ["B", "A"],
  abbreviated: ["BC", "AD"],
  wide: ["Before Christ", "Anno Domini"]
}, kr = {
  narrow: ["1", "2", "3", "4"],
  abbreviated: ["Q1", "Q2", "Q3", "Q4"],
  wide: ["1st quarter", "2nd quarter", "3rd quarter", "4th quarter"]
}, Mr = {
  narrow: ["J", "F", "M", "A", "M", "J", "J", "A", "S", "O", "N", "D"],
  abbreviated: ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"],
  wide: ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"]
}, Pr = {
  narrow: ["S", "M", "T", "W", "T", "F", "S"],
  short: ["Su", "Mo", "Tu", "We", "Th", "Fr", "Sa"],
  abbreviated: ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"],
  wide: ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
}, Rr = {
  narrow: {
    am: "a",
    pm: "p",
    midnight: "mi",
    noon: "n",
    morning: "morning",
    afternoon: "afternoon",
    evening: "evening",
    night: "night"
  },
  abbreviated: {
    am: "AM",
    pm: "PM",
    midnight: "midnight",
    noon: "noon",
    morning: "morning",
    afternoon: "afternoon",
    evening: "evening",
    night: "night"
  },
  wide: {
    am: "a.m.",
    pm: "p.m.",
    midnight: "midnight",
    noon: "noon",
    morning: "morning",
    afternoon: "afternoon",
    evening: "evening",
    night: "night"
  }
}, Wr = {
  narrow: {
    am: "a",
    pm: "p",
    midnight: "mi",
    noon: "n",
    morning: "in the morning",
    afternoon: "in the afternoon",
    evening: "in the evening",
    night: "at night"
  },
  abbreviated: {
    am: "AM",
    pm: "PM",
    midnight: "midnight",
    noon: "noon",
    morning: "in the morning",
    afternoon: "in the afternoon",
    evening: "in the evening",
    night: "at night"
  },
  wide: {
    am: "a.m.",
    pm: "p.m.",
    midnight: "midnight",
    noon: "noon",
    morning: "in the morning",
    afternoon: "in the afternoon",
    evening: "in the evening",
    night: "at night"
  }
}, Yr = function(e, t) {
  var n = Number(e), o = n % 100;
  if (o > 20 || o < 10)
    switch (o % 10) {
      case 1:
        return n + "st";
      case 2:
        return n + "nd";
      case 3:
        return n + "rd";
    }
  return n + "th";
}, Nr = {
  ordinalNumber: Yr,
  era: ie({
    values: Sr,
    defaultWidth: "wide"
  }),
  quarter: ie({
    values: kr,
    defaultWidth: "wide",
    argumentCallback: function(e) {
      return e - 1;
    }
  }),
  month: ie({
    values: Mr,
    defaultWidth: "wide"
  }),
  day: ie({
    values: Pr,
    defaultWidth: "wide"
  }),
  dayPeriod: ie({
    values: Rr,
    defaultWidth: "wide",
    formattingValues: Wr,
    defaultFormattingWidth: "wide"
  })
};
const jr = Nr;
function se(r) {
  return function(e) {
    var t = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, n = t.width, o = n && r.matchPatterns[n] || r.matchPatterns[r.defaultMatchWidth], i = e.match(o);
    if (!i)
      return null;
    var s = i[0], v = n && r.parsePatterns[n] || r.parsePatterns[r.defaultParseWidth], c = Array.isArray(v) ? Ur(v, function(h) {
      return h.test(s);
    }) : Fr(v, function(h) {
      return h.test(s);
    }), g;
    g = r.valueCallback ? r.valueCallback(c) : c, g = t.valueCallback ? t.valueCallback(g) : g;
    var f = e.slice(s.length);
    return {
      value: g,
      rest: f
    };
  };
}
function Fr(r, e) {
  for (var t in r)
    if (r.hasOwnProperty(t) && e(r[t]))
      return t;
}
function Ur(r, e) {
  for (var t = 0; t < r.length; t++)
    if (e(r[t]))
      return t;
}
function $r(r) {
  return function(e) {
    var t = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, n = e.match(r.matchPattern);
    if (!n)
      return null;
    var o = n[0], i = e.match(r.parsePattern);
    if (!i)
      return null;
    var s = r.valueCallback ? r.valueCallback(i[0]) : i[0];
    s = t.valueCallback ? t.valueCallback(s) : s;
    var v = e.slice(o.length);
    return {
      value: s,
      rest: v
    };
  };
}
var Lr = /^(\d+)(th|st|nd|rd)?/i, Ar = /\d+/i, Ir = {
  narrow: /^(b|a)/i,
  abbreviated: /^(b\.?\s?c\.?|b\.?\s?c\.?\s?e\.?|a\.?\s?d\.?|c\.?\s?e\.?)/i,
  wide: /^(before christ|before common era|anno domini|common era)/i
}, qr = {
  any: [/^b/i, /^(a|c)/i]
}, Br = {
  narrow: /^[1234]/i,
  abbreviated: /^q[1234]/i,
  wide: /^[1234](th|st|nd|rd)? quarter/i
}, Hr = {
  any: [/1/i, /2/i, /3/i, /4/i]
}, Vr = {
  narrow: /^[jfmasond]/i,
  abbreviated: /^(jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec)/i,
  wide: /^(january|february|march|april|may|june|july|august|september|october|november|december)/i
}, Gr = {
  narrow: [/^j/i, /^f/i, /^m/i, /^a/i, /^m/i, /^j/i, /^j/i, /^a/i, /^s/i, /^o/i, /^n/i, /^d/i],
  any: [/^ja/i, /^f/i, /^mar/i, /^ap/i, /^may/i, /^jun/i, /^jul/i, /^au/i, /^s/i, /^o/i, /^n/i, /^d/i]
}, Xr = {
  narrow: /^[smtwf]/i,
  short: /^(su|mo|tu|we|th|fr|sa)/i,
  abbreviated: /^(sun|mon|tue|wed|thu|fri|sat)/i,
  wide: /^(sunday|monday|tuesday|wednesday|thursday|friday|saturday)/i
}, Qr = {
  narrow: [/^s/i, /^m/i, /^t/i, /^w/i, /^t/i, /^f/i, /^s/i],
  any: [/^su/i, /^m/i, /^tu/i, /^w/i, /^th/i, /^f/i, /^sa/i]
}, Jr = {
  narrow: /^(a|p|mi|n|(in the|at) (morning|afternoon|evening|night))/i,
  any: /^([ap]\.?\s?m\.?|midnight|noon|(in the|at) (morning|afternoon|evening|night))/i
}, zr = {
  any: {
    am: /^a/i,
    pm: /^p/i,
    midnight: /^mi/i,
    noon: /^no/i,
    morning: /morning/i,
    afternoon: /afternoon/i,
    evening: /evening/i,
    night: /night/i
  }
}, Kr = {
  ordinalNumber: $r({
    matchPattern: Lr,
    parsePattern: Ar,
    valueCallback: function(e) {
      return parseInt(e, 10);
    }
  }),
  era: se({
    matchPatterns: Ir,
    defaultMatchWidth: "wide",
    parsePatterns: qr,
    defaultParseWidth: "any"
  }),
  quarter: se({
    matchPatterns: Br,
    defaultMatchWidth: "wide",
    parsePatterns: Hr,
    defaultParseWidth: "any",
    valueCallback: function(e) {
      return e + 1;
    }
  }),
  month: se({
    matchPatterns: Vr,
    defaultMatchWidth: "wide",
    parsePatterns: Gr,
    defaultParseWidth: "any"
  }),
  day: se({
    matchPatterns: Xr,
    defaultMatchWidth: "wide",
    parsePatterns: Qr,
    defaultParseWidth: "any"
  }),
  dayPeriod: se({
    matchPatterns: Jr,
    defaultMatchWidth: "any",
    parsePatterns: zr,
    defaultParseWidth: "any"
  })
};
const Zr = Kr;
var ea = {
  code: "en-US",
  formatDistance: br,
  formatLong: Cr,
  formatRelative: Er,
  localize: jr,
  match: Zr,
  options: {
    weekStartsOn: 0,
    firstWeekContainsDate: 1
  }
};
const ta = ea;
var ra = /[yYQqMLwIdDecihHKkms]o|(\w)\1*|''|'(''|[^'])+('|$)|./g, aa = /P+p+|P+|p+|''|'(''|[^'])+('|$)|./g, na = /^'([^]*?)'?$/, oa = /''/g, ia = /[a-zA-Z]/;
function it(r, e, t) {
  var n, o, i, s, v, c, g, f, h, D, _, P, d, C, O, E, A, I;
  x(2, arguments);
  var ue = String(e), F = De(), V = (n = (o = t == null ? void 0 : t.locale) !== null && o !== void 0 ? o : F.locale) !== null && n !== void 0 ? n : ta, te = k((i = (s = (v = (c = t == null ? void 0 : t.firstWeekContainsDate) !== null && c !== void 0 ? c : t == null || (g = t.locale) === null || g === void 0 || (f = g.options) === null || f === void 0 ? void 0 : f.firstWeekContainsDate) !== null && v !== void 0 ? v : F.firstWeekContainsDate) !== null && s !== void 0 ? s : (h = F.locale) === null || h === void 0 || (D = h.options) === null || D === void 0 ? void 0 : D.firstWeekContainsDate) !== null && i !== void 0 ? i : 1);
  if (!(te >= 1 && te <= 7))
    throw new RangeError("firstWeekContainsDate must be between 1 and 7 inclusively");
  var Q = k((_ = (P = (d = (C = t == null ? void 0 : t.weekStartsOn) !== null && C !== void 0 ? C : t == null || (O = t.locale) === null || O === void 0 || (E = O.options) === null || E === void 0 ? void 0 : E.weekStartsOn) !== null && d !== void 0 ? d : F.weekStartsOn) !== null && P !== void 0 ? P : (A = F.locale) === null || A === void 0 || (I = A.options) === null || I === void 0 ? void 0 : I.weekStartsOn) !== null && _ !== void 0 ? _ : 0);
  if (!(Q >= 0 && Q <= 6))
    throw new RangeError("weekStartsOn must be between 0 and 6 inclusively");
  if (!V.localize)
    throw new RangeError("locale must contain localize property");
  if (!V.formatLong)
    throw new RangeError("locale must contain formatLong property");
  var J = M(r);
  if (!Qt(J))
    throw new RangeError("Invalid time value");
  var Te = Gt(J), le = Kt(J, Te), $ = {
    firstWeekContainsDate: te,
    weekStartsOn: Q,
    locale: V,
    _originalDate: J
  }, q = ue.match(aa).map(function(Y) {
    var L = Y[0];
    if (L === "p" || L === "P") {
      var G = dr[L];
      return G(Y, V.formatLong);
    }
    return Y;
  }).join("").match(ra).map(function(Y) {
    if (Y === "''")
      return "'";
    var L = Y[0];
    if (L === "'")
      return sa(Y);
    var G = lr[L];
    if (G)
      return !(t != null && t.useAdditionalWeekYearTokens) && gr(Y) && ot(Y, e, String(r)), !(t != null && t.useAdditionalDayOfYearTokens) && hr(Y) && ot(Y, e, String(r)), G(le, Y, V.localize, $);
    if (L.match(ia))
      throw new RangeError("Format string contains an unescaped latin alphabet character `" + L + "`");
    return Y;
  }).join("");
  return q;
}
function sa(r) {
  var e = r.match(na);
  return e ? e[1].replace(oa, "'") : r;
}
function ua(r) {
  x(1, arguments);
  var e = M(r), t = e.getFullYear(), n = e.getMonth(), o = new Date(0);
  return o.setFullYear(t, n + 1, 0), o.setHours(0, 0, 0, 0), o.getDate();
}
function st(r, e) {
  x(2, arguments);
  var t = M(r), n = M(e);
  return t.getTime() > n.getTime();
}
function ut(r, e) {
  x(2, arguments);
  var t = M(r), n = M(e);
  return t.getTime() < n.getTime();
}
function lt(r, e) {
  x(2, arguments);
  var t = M(r), n = M(e);
  return t.getTime() === n.getTime();
}
function la(r, e) {
  x(2, arguments);
  var t = k(e);
  return dt(r, -t);
}
function ca(r, e) {
  x(2, arguments);
  var t = k(e);
  return mt(r, -t);
}
function ye(r) {
  return typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? ye = function(t) {
    return typeof t;
  } : ye = function(t) {
    return t && typeof Symbol == "function" && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t;
  }, ye(r);
}
function ct(r, e) {
  if (x(2, arguments), !e || ye(e) !== "object")
    return new Date(NaN);
  var t = e.years ? k(e.years) : 0, n = e.months ? k(e.months) : 0, o = e.weeks ? k(e.weeks) : 0, i = e.days ? k(e.days) : 0, s = e.hours ? k(e.hours) : 0, v = e.minutes ? k(e.minutes) : 0, c = e.seconds ? k(e.seconds) : 0, g = ca(r, n + t * 12), f = la(g, i + o * 7), h = v + s * 60, D = c + h * 60, _ = D * 1e3, P = new Date(f.getTime() - _);
  return P;
}
const da = ({ initialRange: r, onUpdate: e }) => {
  const [t, n] = ae([]), [o, i] = ae([]), [s, v] = ae(new Date().getMonth()), [c, g] = ae(new Date().getFullYear()), [f, h] = ae({
    startDate: (r == null ? void 0 : r.startDate) ?? new Date(),
    endDate: (r == null ? void 0 : r.endDate) ?? j(new Date(), { days: 20 })
  }), D = (d, C, O, E = "()") => {
    if (!["()", "[]", "(]", "[)"].includes(E))
      throw new Error("Inclusivity parameter must be one of (), [], (], [)");
    const A = E[0] === "[", I = E[1] === "]";
    return (A && lt(C, d) || ut(C, d)) && (I && lt(O, d) || st(O, d));
  }, _ = ({
    month: d,
    year: C
  }) => {
    const O = zt(new Date(C, d));
    Jt(new Date(C, d));
    const E = O.getDay(), A = ua(O), I = [...Array(A).keys()].map((F) => F + 1);
    return [...Array(42).keys()].map((F) => F < E || F > A + E - 1 ? null : I[F - E]);
  };
  Ke(() => {
    const d = _({
      month: new Date(c, s).getMonth(),
      year: new Date(c, s).getFullYear()
    });
    n(d);
    const C = _({
      month: j(new Date(c, s), { months: 1 }).getMonth(),
      year: j(new Date(c, s), { months: 1 }).getFullYear()
    });
    i(C);
  }, [s, c]), Ke(() => {
    f.startDate && f.endDate && (e == null || e(f));
  }, [f]);
  const P = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
  return /* @__PURE__ */ b.jsxs("div", { className: "flex", children: [
    /* @__PURE__ */ b.jsxs("div", { className: "drop-shadow-sm shadow-sm min-w-[10rem] p-3 rounded-lg border border-r-0 flex flex-col", children: [
      /* @__PURE__ */ b.jsxs("div", { className: "flex justify-between items-center", children: [
        /* @__PURE__ */ b.jsxs("h3", { className: "text-lg", children: [
          it(new Date(c, s), "MMMM"),
          " ",
          new Date(c, s).getFullYear()
        ] }),
        /* @__PURE__ */ b.jsxs("div", { className: "flex gap-x-2 items-center", children: [
          /* @__PURE__ */ b.jsx(
            "button",
            {
              type: "button",
              className: "p-2 rounded-lg hover:bg-gray-300 border drop-shadow-sm",
              onClick: () => {
                const d = ct(new Date(c, s), {
                  months: 1
                });
                v(d.getMonth()), g(d.getFullYear());
              },
              children: /* @__PURE__ */ b.jsx(
                "svg",
                {
                  xmlns: "http://www.w3.org/2000/svg",
                  fill: "none",
                  viewBox: "0 0 24 24",
                  strokeWidth: 1.5,
                  stroke: "currentColor",
                  className: "w-4 h-4",
                  children: /* @__PURE__ */ b.jsx(
                    "path",
                    {
                      strokeLinecap: "round",
                      strokeLinejoin: "round",
                      d: "M15.75 19.5L8.25 12l7.5-7.5"
                    }
                  )
                }
              )
            }
          ),
          /* @__PURE__ */ b.jsx(
            "button",
            {
              type: "button",
              className: "p-2 rounded-lg hover:bg-gray-300 border drop-shadow-sm",
              onClick: () => {
                const d = j(new Date(c, s), {
                  months: 1
                });
                v(d.getMonth()), g(d.getFullYear());
              },
              children: /* @__PURE__ */ b.jsx(
                "svg",
                {
                  xmlns: "http://www.w3.org/2000/svg",
                  fill: "none",
                  viewBox: "0 0 24 24",
                  strokeWidth: 1.5,
                  stroke: "currentColor",
                  className: "w-4 h-4",
                  children: /* @__PURE__ */ b.jsx(
                    "path",
                    {
                      strokeLinecap: "round",
                      strokeLinejoin: "round",
                      d: "M8.25 4.5l7.5 7.5-7.5 7.5"
                    }
                  )
                }
              )
            }
          )
        ] })
      ] }),
      /* @__PURE__ */ b.jsx("div", { className: "grid grid-cols-7 p-3 gap-2 mt-3", children: P.map((d, C) => /* @__PURE__ */ b.jsx("div", { className: "text-center w-12", children: d }, C)) }),
      /* @__PURE__ */ b.jsx("div", { className: "grid grid-cols-7 p-3 gap-2", children: t.map(
        (d, C) => d ? /* @__PURE__ */ b.jsx(
          "button",
          {
            type: "button",
            className: `rounded-lg border flex w-12 justify-center p-2 hover:bg-blue-600 hover:text-white ${Z(
              new Date(c, s, d),
              f.startDate
            ) ? (
              // dateRange.startDate?.isSame(
              //   moment({ day: v, month: currMonth, year: currYear }),
              //   "day"
              // )
              "bg-blue-600 text-white"
            ) : ""} ${Z(
              new Date(c, s, d),
              new Date()
            ) ? (
              // moment({ day: v, month: currMonth, year: currYear }).isSame(
              //   moment(),
              //   "day"
              // )
              "ring-blue-400 ring-2"
            ) : ""} ${Z(
              new Date(c, s, d),
              f.endDate
            ) ? (
              // dateRange.endDate?.isSame(
              //   moment({ day: v, month: currMonth, year: currYear }),
              //   "day"
              // )
              "bg-blue-600 text-white"
            ) : ""} ${D(
              new Date(c, s, d),
              f.startDate,
              f.endDate,
              "[]"
            ) ? (
              // moment({
              //   day: v,
              //   month: currMonth,
              //   year: currYear,
              // }).isBetween(
              //   dateRange.startDate,
              //   dateRange.endDate,
              //   "day",
              //   "()"
              // )
              "bg-blue-200"
            ) : ""}`,
            onClick: () => {
              f.startDate || h((O) => ({
                ...O,
                startDate: new Date(c, s, d)
                // moment({
                //   day: v,
                //   month: currMonth,
                //   year: currYear,
                // }),
              })), f.startDate && !f.endDate && h((O) => ({
                ...O,
                endDate: new Date(c, s, d)
                // moment({
                //   day: v,
                //   month: currMonth,
                //   year: currYear,
                // }),
              })), f.startDate && f.endDate && h({
                startDate: new Date(c, s, d),
                // moment({
                //   day: v,
                //   month: currMonth,
                //   year: currYear,
                // }),
                endDate: null
              });
            },
            children: d
          },
          C
        ) : /* @__PURE__ */ b.jsx("span", {}, C)
      ) })
    ] }),
    /* @__PURE__ */ b.jsxs("div", { className: "drop-shadow-sm shadow-sm min-w-[10rem] p-3 rounded-lg border flex flex-col", children: [
      /* @__PURE__ */ b.jsxs("div", { className: "flex justify-between items-center", children: [
        /* @__PURE__ */ b.jsxs("h3", { className: "text-lg", children: [
          it(
            j(new Date(c, s), { months: 1 }),
            "MMMM"
          ),
          " ",
          j(new Date(c, s), { years: 1 }).getFullYear()
        ] }),
        /* @__PURE__ */ b.jsxs("div", { className: "flex gap-x-2 items-center", children: [
          /* @__PURE__ */ b.jsx(
            "button",
            {
              type: "button",
              className: "p-2 rounded-lg hover:bg-gray-300 border drop-shadow-sm",
              onClick: () => {
                const d = ct(new Date(c, s), {
                  months: 1
                });
                v(d.getMonth()), g(d.getFullYear());
              },
              children: /* @__PURE__ */ b.jsx(
                "svg",
                {
                  xmlns: "http://www.w3.org/2000/svg",
                  fill: "none",
                  viewBox: "0 0 24 24",
                  strokeWidth: 1.5,
                  stroke: "currentColor",
                  className: "w-4 h-4",
                  children: /* @__PURE__ */ b.jsx(
                    "path",
                    {
                      strokeLinecap: "round",
                      strokeLinejoin: "round",
                      d: "M15.75 19.5L8.25 12l7.5-7.5"
                    }
                  )
                }
              )
            }
          ),
          /* @__PURE__ */ b.jsx(
            "button",
            {
              type: "button",
              className: "p-2 rounded-lg hover:bg-gray-300 border drop-shadow-sm",
              onClick: () => {
                const d = j(new Date(c, s), {
                  months: 1
                });
                v(d.getMonth()), g(d.getFullYear());
              },
              children: /* @__PURE__ */ b.jsx(
                "svg",
                {
                  xmlns: "http://www.w3.org/2000/svg",
                  fill: "none",
                  viewBox: "0 0 24 24",
                  strokeWidth: 1.5,
                  stroke: "currentColor",
                  className: "w-4 h-4",
                  children: /* @__PURE__ */ b.jsx(
                    "path",
                    {
                      strokeLinecap: "round",
                      strokeLinejoin: "round",
                      d: "M8.25 4.5l7.5 7.5-7.5 7.5"
                    }
                  )
                }
              )
            }
          )
        ] })
      ] }),
      /* @__PURE__ */ b.jsx("div", { className: "grid grid-cols-7 p-3 gap-2 mt-3", children: P.map((d, C) => /* @__PURE__ */ b.jsx("div", { className: "text-center w-12", children: d }, C)) }),
      /* @__PURE__ */ b.jsx("div", { className: "grid grid-cols-7 p-3 gap-2", children: o.map(
        (d, C) => d ? /* @__PURE__ */ b.jsx(
          "button",
          {
            type: "button",
            className: `rounded-lg border flex w-12 justify-center p-2 hover:bg-blue-600 hover:text-white ${Z(
              j(new Date(c, s, d), {
                months: 1
              }),
              f.startDate
            ) ? (
              // dateRange.startDate?.isSame(
              //   moment({
              //     day: v,
              //     month: moment({
              //       day: v,
              //       month: currMonth,
              //       year: currYear,
              //     })
              //       .add(1, "month")
              //       .month(),
              //     year: moment({ day: v, month: currMonth, year: currYear })
              //       .add(1, "month")
              //       .year(),
              //   }),
              //   "day"
              // )
              "bg-blue-600 text-white"
            ) : ""} ${Z(
              j(new Date(c, s, d), {
                months: 1
              }),
              new Date()
            ) ? (
              // moment({
              //   day: v,
              //   month: moment({ day: v, month: currMonth, year: currYear })
              //     .add(1, "month")
              //     .month(),
              //   year: moment({ day: v, month: currMonth, year: currYear })
              //     .add(1, "month")
              //     .year(),
              // }).isSame(moment(), "day")
              "ring-blue-400 ring-2"
            ) : ""} ${Z(
              j(new Date(c, s, d), {
                months: 1
              }),
              f.endDate
            ) ? (
              // dateRange.endDate?.isSame(
              //   moment({
              //     day: v,
              //     month: moment({
              //       day: v,
              //       month: currMonth,
              //       year: currYear,
              //     })
              //       .add(1, "month")
              //       .month(),
              //     year: moment({ day: v, month: currMonth, year: currYear })
              //       .add(1, "month")
              //       .year(),
              //   }),
              //   "day"
              // )
              "bg-blue-600 text-white"
            ) : ""} ${// isBetween(new Date(currYear, currMonth, v), {}))
            D(
              j(new Date(c, s, d), {
                months: 1
              }),
              f.startDate,
              f.endDate,
              "[]"
            ) ? (
              // moment({
              //   day: v,
              //   month: moment({ day: v, month: currMonth, year: currYear })
              //     .add(1, "month")
              //     .month(),
              //   year: moment({ day: v, month: currMonth, year: currYear })
              //     .add(1, "month")
              //     .year(),
              // }).isBetween(
              //   dateRange.startDate,
              //   dateRange.endDate,
              //   "day",
              //   "()"
              // )
              "bg-blue-200"
            ) : ""}`,
            onClick: () => {
              f.startDate || h((O) => ({
                ...O,
                startDate: j(new Date(c, s, d), {
                  months: 1
                })
              })), f.startDate && !f.endDate && h((O) => ({
                ...O,
                endDate: j(new Date(c, s, d), {
                  months: 1
                })
                // endDate: moment({
                //   day: v,
                //   month: moment({
                //     day: v,
                //     month: currMonth,
                //     year: currYear,
                //   })
                //     .add(1, "month")
                //     .month(),
                //   year: moment({
                //     day: v,
                //     month: currMonth,
                //     year: currYear,
                //   })
                //     .add(1, "month")
                //     .year(),
                // }),
              })), f.startDate && f.endDate && h({
                startDate: j(new Date(c, s, d), {
                  months: 1
                }),
                // moment({
                //   day: v,
                //   month: moment({
                //     day: v,
                //     month: currMonth,
                //     year: currYear,
                //   })
                //     .add(1, "month")
                //     .month(),
                //   year: moment({
                //     day: v,
                //     month: currMonth,
                //     year: currYear,
                //   })
                //     .add(1, "month")
                //     .year(),
                // }),
                endDate: null
              });
            },
            children: d
          },
          C
        ) : /* @__PURE__ */ b.jsx("span", {}, C)
      ) })
    ] })
  ] });
};
export {
  da as default
};
